fuel-relayer/src/pending_blocks.rs

```diff
@@ -0,0 +1,617 @@
+use anyhow::Error;
+use ethers_core::{
+    k256::ecdsa::SigningKey,
+    types::{TransactionRequest, H160, U256},
+};
+use ethers_middleware::{
+    gas_escalator::{Frequency, GasEscalatorMiddleware, GeometricGasPrice},
+    NonceManagerMiddleware, SignerMiddleware,
+};
+use ethers_providers::Middleware;
+use fuel_core_interfaces::{
+    model::{BlockHeight, DaBlockHeight, SealedFuelBlock},
+    relayer::RelayerDb,
+};
+use fuel_tx::Bytes32;
+use std::{cmp::max, collections::VecDeque, sync::Arc};
+
+// use the ethers_signers crate to manage LocalWallet and Signer
+use crate::abi;
+use ethers_signers::{LocalWallet, Signer};
+use tracing::{debug, error, info, warn};
+
+/// Pending Fuel Blocks waiting to be finalized inside client. Until then
+/// there is possibility that they are going to be reverted
+pub struct PendingBlocks {
+    signer: LocalWallet,
+    contract_address: H160,
+    // Pending block commits seen on DA layer and waiting to be finalized
+    pending_block_commits: VecDeque<PendingBlock>,
+    // Highest known chain height, used to check if we are seeing lag between block commits and our fule chain
+    chain_height: BlockHeight,
```

Comment:
Is this the same thing as `last_commited_finalized_fuel_height`, but includes the best height seen even in pending commitments? This name is way too ambiguous for me. I shouldn't have to read this comment each time to know what this is.

```suggestion
    best_pending_chain_height_commit: BlockHeight,
```

Reply:
It is `Highest known chain height`